apollo/handlers/user.py

```diff
@@ -43,12 +43,25 @@ def post_user(user_data: CreateUserSchema,
 def patch_user(user_data: UpdateUserSchema, request: Request,
                session: Session = Depends(get_session)):
     user = request.current_user
-    data = user_data.dict()
+    data = user_data.dict(exclude_unset=True)
+
+    if data.get('password'):
+        user, data, error = update_user_password_and_data(user, data)
+        if error:
+            return error
+
+    user.set_fields(data)
+    saved_user, _ = save(session, user)
+    return saved_user
+
 
-    if not compare_plaintext_to_hash(user_data.old_password,
+def update_user_password_and_data(user, data):
+    error = None
+    print(data)
+    if not compare_plaintext_to_hash(data['old_password'],
                                      user.password_hash,
                                      user.password_salt):
-        return JSONResponse(
+        error = JSONResponse(
```

Comment:
This should be raised. You probably can't use this class but instead `HTTPException` or one of the new exception classes (might need modification).

Reply:
>This should be raised. You probably can't use this class but instead HTTPException or one of the new exception classes (might need modification).